joomla/joomla-cms
# com_fields : checking for availability of component, correcting 2 fatal errors

Pull Request for Issue https://github.com/joomla/joomla-cms/issues/12645
and for issue https://github.com/joomla/joomla-cms/issues/12633
### Summary of Changes

Prevents fields menu and sidebar from displaying when com_fields is not enabled.
Displays a Warning if the fields system plugin is disabled.
Corrected 2 fatal errors JFolder and JFile not found when saving a custom field.
Alpha ordered and modified the xml_description language string for the system plugin as the custom fields will NOT display if disabled.
### Testing Instructions
Patch
Disable the component com_fields in Extensions Manager=>Manage 
result: menus and sidebar do not show (Users, contacts, content)

Then enable it again and disable the fields system plugin. Display the fields manager
Result : a Warning is displayed in the manager
![screen shot 2016-10-31 at 08 03 22](https://cloud.githubusercontent.com/assets/869724/19846946/8243c068-9f43-11e6-959d-fbe1a1da4320.png)


Create a new custom field and save. Before patch fatal error, after, it saves fine

TODO: change the fields manager and fields group manager title to fit the component concerned
